src/ILLink.RoslynAnalyzer/INamedTypeSymbolExtensions.cs

```diff
@@ -28,5 +29,31 @@ internal static bool HasName (this INamedTypeSymbol type, string typeName)
 
 			return true;
 		}
+
+		internal static IEnumerable<(ISymbol InterfaceMember, ISymbol ImplementationMember)> GetMemberInterfaceImplementationPairs (this INamedTypeSymbol namedType)
+		{
+			var interfaces = namedType.Interfaces;
+			foreach (INamedTypeSymbol iface in interfaces) {
+				foreach (var pair in GetMatchingMembers (namedType, iface)) {
+					yield return pair;
+				}
+			}
+		}
+
+		private static IEnumerable<(ISymbol InterfaceMember, ISymbol ImplementationMember)> GetMatchingMembers (INamedTypeSymbol implementationSymbol, INamedTypeSymbol interfaceSymbol)
+		{
+			var members = interfaceSymbol.GetMembers ();
+			foreach (var member in members) {
+				if (member is ISymbol interfaceMember
```

Comment:
nit: could we replace the `ISymbol` check by doing `foreach (ISymbol interfaceMember in interfaceSymbol.GetMembers ())`?